### ISSUE TITLE
Need to align the behavior when Navigation app started with Bluetooth transport

### ISSUE THREAD

**Issue by shiniwat** [smartdevicelink/sdl_java_suite#893]
### Bug Report
Need to align the behavior when Navigation app started with Bluetooth transport

##### Reproduction Steps
1. Prerequisite: SDL Core is lower than 5.0 (i.e. 4.5 or lower). Bluetooth connection is established.
2. Without connecting with USB, start Navigation app, whose MultiplexTransportConfig has requiresHighBandwidth = true.

##### Expected Behavior
Even though the app has requiresHighBandwidth flag set, the connection with SDLCore should work and RegisterAppInterface should work with Bluetooth transport.

##### Observed Behavior
When starting protocol session, SDL Proxy refuses the connection, and eventually proxy is closed.

##### OS & Version Information
* Android Version: Android 7.0
* SDL Android Version: commit [13e04232](13e04232a32ba311e0d3a7847b7f1f841f6f10ec) from release/4.7.0-rc1 branch
* Testing Against: running with SDLCore 4.5, generic HMI.

##### Test Case, Sample Code, and / or Example App
I thought this behavior would be OK when running with older SDLCore that does not support secondary transport. However when I looked into how iOS SDL Proxy works, it works in the following way.
a) Suppose iAP is connected with Bluetooth. When proxy gets started on Navigation app, it works normally; i.e. RegisterAppInterface works on iAP (BT).
b) When the app gets HMI_FULL state, the app checks to see hmiCapability, which is returned in OnRegisterAppInterfaceResponse, and if videoStreaming is false, the app won't start video streaming; instead, it shows "Please connect USB cable" message on HU (e.g. by using non-Media template).
c) When USB is connected, iOS proxy automatically restarts with refreshed EA session (USB), and then  performs RegisterAppInterface. The hmiCapability now returns videoStreaming = true, then the app starts video streaming when it gets HMI_FULL status.

I think we should do the same thing on Android, so that the UX would be aligned with iOS. If running with SDLCore 5.0, RegisterAppInterface works with BT already, so this should be OK.

In order to do this, all I needed to do in SDL Proxy is just commenting out some lines in SdlProtocol.java, something like:
```
                    if (requiresHighBandwidth
                            && TransportType.BLUETOOTH.equals(transportRecord.getType())) {
                        //transport can't support high bandwidth, but we allow apps to register with BT
                        //onTransportNotAccepted(transportRecord.getType() + " can't support high bandwidth requirement, and secondary transport not supported in this protocol version: " + version);
                        //return;
                    }
```
In addition to changing the proxy, we probably should update a guideline for developer.

**Comment by joeygrover** [comment]
I disagree with this implementation. It should be up to the developer if they wish to connect while over bluetooth or not, hence the `requiresHighBandwidth` flag. If they set that to false, then they have the opportunity to display any information they want while connected to bluetooth. When USB is connected they will be informed and be able to make a choice. They proposed fix will not result in the behavior desired as the proxy is much more complicated when connected to the router service for the USB transport. Also simply commenting out lines is not a correct way to change behavior. 

We should not decide an app's user experience for them. I believe iOS should also have an issue raised as it is possible that a core implementation does not support video streaming while the USB cord is plugged in and therefore the message of "Please connect USB cable" does not make sense.

**Comment by joeygrover** [comment]
This would require a major reworking of how the proxy is managed and therefore require an evolution proposal. At this point there is no way it can be included in this next release (4.7) which will mean it will also become a breaking change as it will alter what the developer can expect from how the library handles the said situation. 

A developer can use the flags and and the `MultiplexConfigTransport.TransportListener` to find out when USB becomes available if they wish to stay connected while bluetooth is enabled.

The only reason iOS stays connected is because there is no way to tell the difference between a bluetooth or USB EASession on the app's side.
